test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -256,6 +256,55 @@ main() {
               reason: 'should preserve existing inheritance');
         });
 
+        test('with static PropsMeta declaration', () {
+          final transformedLine = 'static const PropsMeta meta = \$Props(FooProps);';
+
+          setUpAndGenerate('''
+           @Factory()
+           UiFactory<FooProps> Foo;
+        
+           @Props()
+           class FooProps {
+            static const PropsMeta meta = \$metaForFooProps;
+           }
+           
+           @Component()
+           class FooComponent {
+            render() => null;
+           }
+           '''
+          );
+
+          var transformedSource = transformedFile.getTransformedText();
+          expect(transformedSource, contains(transformedLine));
```

Comment:
I think it might make sense to also verify that the original meta line is not longer present to ensure that the transformer is replacing and not simply adding a line.